packages/handlersjs-http/lib/services/sync.service.spec.ts

```diff
@@ -0,0 +1,288 @@
+jest.mock('node-fetch');
+
+import { MemoryStore } from '@digita-ai/handlersjs-core';
+import fetch from 'node-fetch';
+import { SyncService } from './sync.service';
+const { Response } = jest.requireActual('node-fetch');
+
+describe('SyncService', () => {
+
+  const headers = new Headers({
+    'Content-Type': 'application/json',
+    Accept: '*/*',
+  });
+
+  const peers = new Set([ 'peer1.com', 'peer2.com' ]);
+
+  type M = {
+    storage: Set<number>; } & {
+    peers: Set<string>;
+    [key: string]: any;
+  };
+
+  let store: MemoryStore<M>;
+  let syncService: SyncService<number, 'storage', 'peers', M>;
+  let fetchMock: jest.MockedFunction<typeof fetch>;
+
+  // saves the latest "modified since" header from mocked requests
+  let latestModifiedSince: string | undefined;
+
+  beforeEach(() => {
+
+    jest.resetAllMocks();
+
+    fetchMock = (fetch as jest.MockedFunction<typeof fetch>);
+    latestModifiedSince = undefined;
+
+    fetchMock.mockImplementation(async (url, options) => {
+
+      latestModifiedSince = options?.headers['If-Modified-Since'];
+
+      return new Response('[]', { status: 200, headers });
+
+    });
```

Comment:
Is this the easiest way? Can't we just check wether the mock function is called with an options object containing that header?

Reply:
since I need this value in multiple places it seemed like the simplest way, since the method needs to be mocked anyways. The other option would be making a function or doing it in-place everytime, which would look like something this: 

`const latestModifiedSince = fetchMock.mock.calls[fetchMock.mock.calls.length - 1][1]?.headers['If-Modified-Since'];`

Reply:
I was thinking more about something using [fetch-mock](http://www.wheresrhys.co.uk/fetch-mock/)'s `called` function with a matcher for the header.

Reply:
it is used in multiple places, in which some don't directly place an assertion on it but rather extract the value, do something with it, and do an assertion on something derived from it

I removed the field now and included a function to extract it